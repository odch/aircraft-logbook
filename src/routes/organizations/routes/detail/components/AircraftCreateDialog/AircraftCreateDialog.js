import React from 'react'
import PropTypes from 'prop-types'
import { injectIntl, FormattedMessage } from 'react-intl'
import { withStyles } from '@material-ui/core/styles'
import Button from '@material-ui/core/Button'
import TextField from '@material-ui/core/TextField'
import Dialog from '@material-ui/core/Dialog'
import DialogActions from '@material-ui/core/DialogActions'
import DialogContent from '@material-ui/core/DialogContent'
import DialogContentText from '@material-ui/core/DialogContentText'
import DialogTitle from '@material-ui/core/DialogTitle'
import CircularProgress from '@material-ui/core/CircularProgress'
import FormControl from '@material-ui/core/FormControl'
import FormHelperText from '@material-ui/core/FormHelperText'
import { intl as intlShape } from '../../../../../../shapes'

const styles = {
  loadingIndicator: {
    marginRight: 5
  }
}

class AircraftCreateDialog extends React.Component {
  handleRegistrationChange = e => {
    this.props.updateData({
      registration: e.target.value.toUpperCase().replace(/[^A-Z0-9-]/g, '')
    })
  }

  handleSubmit = e => {
    const { organizationId, data, onSubmit } = this.props

    e.preventDefault()

    if (onSubmit) {
      onSubmit(organizationId, data)
    }
  }

  handleClose = () => {
    const { submitted, onClose } = this.props

    if (!submitted && onClose) {
      onClose()
    }
  }

  render() {
    const { data, submitted, duplicate, classes, onClose } = this.props
    return (
      <Dialog onClose={this.handleClose} data-cy="aircraft-create-dialog" open>
        <DialogTitle>
          <FormattedMessage id="aircrafts.create.dialog.title" />
        </DialogTitle>
        <form onSubmit={this.handleSubmit}>
          <DialogContent>
            <DialogContentText>
              <FormattedMessage id="aircrafts.create.dialog.text" />
            </DialogContentText>
            <FormControl fullWidth error={duplicate}>
              <TextField
                autoFocus
                margin="dense"
                label={this.props.intl.formatMessage({
                  id: 'aircrafts.create.dialog.registration'
                })}
                type="text"
                fullWidth
                required
                value={data.registration}
                onChange={this.handleRegistrationChange}
                data-cy="registration-field"
                disabled={submitted}
              />
              {duplicate && (
                <FormHelperText>
                  <FormattedMessage id="aircrafts.create.dialog.duplicate" />
                </FormHelperText>
              )}
            </FormControl>
          </DialogContent>
          <DialogActions>
            <Button onClick={onClose} color="primary" disabled={submitted}>
              <FormattedMessage id="aircrafts.create.dialog.buttons.cancel" />
            </Button>
            <Button
              type="submit"
              color="primary"
              data-cy="create-button"
              disabled={submitted}
            >
              {submitted && (
                <CircularProgress
                  size={16}
                  className={classes.loadingIndicator}
                />
              )}
              <FormattedMessage id="aircrafts.create.dialog.buttons.create" />
            </Button>
          </DialogActions>
        </form>
      </Dialog>
    )
  }
}

AircraftCreateDialog.propTypes = {
  organizationId: PropTypes.string.isRequired,
  data: PropTypes.shape({
    registration: PropTypes.string
  }).isRequired,
  submitted: PropTypes.bool,
  duplicate: PropTypes.bool,
  classes: PropTypes.object.isRequired,
  onClose: PropTypes.func,
  onSubmit: PropTypes.func,
  updateData: PropTypes.func.isRequired,
  intl: intlShape
}

export default withStyles(styles)(injectIntl(AircraftCreateDialog))
