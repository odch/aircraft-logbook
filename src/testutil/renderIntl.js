import React from 'react'
import renderer from 'react-test-renderer'
import { createMount } from '@material-ui/core/test-utils'
import { IntlProvider } from 'react-intl'
import pretty from 'pretty'
import messages from '../messages'

const LOCALE = 'de'

const renderInIntlProvider = component => (
  <IntlProvider locale={LOCALE} messages={messages[LOCALE]}>
    {component}
  </IntlProvider>
)

const renderIntl = (component, options) =>
  renderer.create(renderInIntlProvider(component), options)

export const renderIntlUpdate = (renderer, component, options) =>
  renderer.update(renderInIntlProvider(component), options)

export const renderIntlMaterial = (component, returnPrettyHtml = false) => {
  const rendered = createMount()(renderInIntlProvider(component))
  return returnPrettyHtml ? pretty(rendered.html()) : rendered
}

export default renderIntl
