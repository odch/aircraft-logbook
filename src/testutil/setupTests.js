import * as enzyme from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import moment from 'moment'
import momentTimezone from 'moment-timezone'
import { init as initFeatureToggles } from '../util/featureToggles'

enzyme.configure({ adapter: new Adapter() })

moment.locale('de')
momentTimezone.locale('de')

initFeatureToggles()
