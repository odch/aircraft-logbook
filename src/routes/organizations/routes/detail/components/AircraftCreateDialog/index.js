import AircraftCreateDialog from './AircraftCreateDialog'

export default AircraftCreateDialog
