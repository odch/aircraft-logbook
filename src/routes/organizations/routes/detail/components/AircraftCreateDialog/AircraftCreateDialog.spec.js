import React from 'react'
import { renderIntlMaterial } from '../../../../../../testutil/renderIntl'
import AircraftCreateDialog from './AircraftCreateDialog'

describe('routes', () => {
  describe('organizations', () => {
    describe('routes', () => {
      describe('detail', () => {
        describe('components', () => {
          describe('AircraftCreateDialog', () => {
            it('renders correctly', () => {
              const renderedValue = renderIntlMaterial(
                <AircraftCreateDialog
                  data={{ registration: 'HB-ABC' }}
                  updateData={() => {}}
                />,
                true
              )
              expect(renderedValue).toMatchSnapshot()
            })

            it('renders with duplicate error state', () => {
              const renderedValue = renderIntlMaterial(
                <AircraftCreateDialog
                  data={{ registration: 'HB-ABC' }}
                  updateData={() => {}}
                  duplicate
                />,
                true
              )
              expect(renderedValue).toMatchSnapshot()
            })
          })
        })
      })
    })
  })
})
