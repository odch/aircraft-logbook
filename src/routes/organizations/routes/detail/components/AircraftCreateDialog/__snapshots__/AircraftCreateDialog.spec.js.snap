// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`routes organizations routes detail components AircraftCreateDialog renders correctly 1`] = `
"<div role=\\"presentation\\" class=\\"MuiDialog-root\\" data-cy=\\"aircraft-create-dialog\\" style=\\"position: fixed; z-index: 1300; right: 0px; bottom: 0px; top: 0px; left: 0px;\\">
  <div class=\\"MuiBackdrop-root\\" aria-hidden=\\"true\\" style=\\"opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;\\"></div>
  <div tabindex=\\"0\\" data-test=\\"sentinelStart\\"></div>
  <div class=\\"MuiDialog-container MuiDialog-scrollPaper\\" style=\\"opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;\\" role=\\"document\\" tabindex=\\"-1\\">
    <div class=\\"MuiPaper-root MuiPaper-elevation24 MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthSm MuiPaper-rounded\\" role=\\"dialog\\">
      <div class=\\"MuiDialogTitle-root\\">
        <h2 class=\\"MuiTypography-root MuiTypography-h6\\">Neues Flugzeug erstellen</h2>
      </div>
      <form>
        <div class=\\"MuiDialogContent-root\\">
          <p class=\\"MuiTypography-root MuiDialogContentText-root MuiTypography-body1 MuiTypography-colorTextSecondary\\">Trage im folgenden Feld die Immatrikulation deines Flugzeugs ein.</p>
          <div class=\\"MuiFormControl-root MuiFormControl-fullWidth\\">
            <div class=\\"MuiFormControl-root MuiTextField-root MuiFormControl-marginDense MuiFormControl-fullWidth\\" data-cy=\\"registration-field\\"><label class=\\"MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-marginDense MuiFormLabel-filled Mui-focused Mui-focused Mui-required Mui-required\\" data-shrink=\\"true\\">Immatrikulation<span class=\\"MuiFormLabel-asterisk MuiInputLabel-asterisk\\"> *</span></label>
              <div class=\\"MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth Mui-focused Mui-focused MuiInputBase-formControl MuiInput-formControl MuiInputBase-marginDense\\"><input aria-invalid=\\"false\\" class=\\"MuiInputBase-input MuiInput-input MuiInputBase-inputMarginDense MuiInput-inputMarginDense\\" required=\\"\\" type=\\"text\\" value=\\"HB-ABC\\"></div>
            </div>
          </div>
        </div>
        <div class=\\"MuiDialogActions-root MuiDialogActions-spacing\\"><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiButton-label\\">Abbrechen</span><span class=\\"MuiTouchRipple-root\\"></span></button><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary\\" tabindex=\\"0\\" type=\\"submit\\" data-cy=\\"create-button\\"><span class=\\"MuiButton-label\\">Erstellen</span><span class=\\"MuiTouchRipple-root\\"></span></button></div>
      </form>
    </div>
  </div>
  <div tabindex=\\"0\\" data-test=\\"sentinelEnd\\"></div>
</div>"
`;

exports[`routes organizations routes detail components AircraftCreateDialog renders with duplicate error state 1`] = `
"<div role=\\"presentation\\" class=\\"MuiDialog-root\\" data-cy=\\"aircraft-create-dialog\\" style=\\"position: fixed; z-index: 1300; right: 0px; bottom: 0px; top: 0px; left: 0px;\\">
  <div class=\\"MuiBackdrop-root\\" aria-hidden=\\"true\\" style=\\"opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;\\"></div>
  <div tabindex=\\"0\\" data-test=\\"sentinelStart\\"></div>
  <div class=\\"MuiDialog-container MuiDialog-scrollPaper\\" style=\\"opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;\\" role=\\"document\\" tabindex=\\"-1\\">
    <div class=\\"MuiPaper-root MuiPaper-elevation24 MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthSm MuiPaper-rounded\\" role=\\"dialog\\">
      <div class=\\"MuiDialogTitle-root\\">
        <h2 class=\\"MuiTypography-root MuiTypography-h6\\">Neues Flugzeug erstellen</h2>
      </div>
      <form>
        <div class=\\"MuiDialogContent-root\\">
          <p class=\\"MuiTypography-root MuiDialogContentText-root MuiTypography-body1 MuiTypography-colorTextSecondary\\">Trage im folgenden Feld die Immatrikulation deines Flugzeugs ein.</p>
          <div class=\\"MuiFormControl-root MuiFormControl-fullWidth\\">
            <div class=\\"MuiFormControl-root MuiTextField-root MuiFormControl-marginDense MuiFormControl-fullWidth\\" data-cy=\\"registration-field\\"><label class=\\"MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-marginDense MuiFormLabel-filled Mui-focused Mui-focused Mui-required Mui-required\\" data-shrink=\\"true\\">Immatrikulation<span class=\\"MuiFormLabel-asterisk MuiInputLabel-asterisk\\"> *</span></label>
              <div class=\\"MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth Mui-focused Mui-focused MuiInputBase-formControl MuiInput-formControl MuiInputBase-marginDense\\"><input aria-invalid=\\"false\\" class=\\"MuiInputBase-input MuiInput-input MuiInputBase-inputMarginDense MuiInput-inputMarginDense\\" required=\\"\\" type=\\"text\\" value=\\"HB-ABC\\"></div>
            </div>
            <p class=\\"MuiFormHelperText-root Mui-error\\">Es existiert bereits ein Flugzeug mit dieser Immatrikulation.</p>
          </div>
        </div>
        <div class=\\"MuiDialogActions-root MuiDialogActions-spacing\\"><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiButton-label\\">Abbrechen</span><span class=\\"MuiTouchRipple-root\\"></span></button><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary\\" tabindex=\\"0\\" type=\\"submit\\" data-cy=\\"create-button\\"><span class=\\"MuiButton-label\\">Erstellen</span><span class=\\"MuiTouchRipple-root\\"></span></button></div>
      </form>
    </div>
  </div>
  <div tabindex=\\"0\\" data-test=\\"sentinelEnd\\"></div>
</div>"
`;
